browser/brave_shields/brave_shields_web_contents_observer.cc

```diff
@@ -278,17 +312,8 @@ void BraveShieldsWebContentsObserver::ReadyToCommitNavigation(
     }
   }
 
-  navigation_handle->GetWebContents()->ForEachRenderFrameHost(
-      [this](content::RenderFrameHost* rfh) {
-        GetBraveShieldsRemote(rfh)->SetAllowScriptsFromOriginsOnce(
-            allowed_scripts_);
-        if (content::BrowserContext* context = rfh->GetBrowserContext()) {
-          if (PrefService* pref_service = user_prefs::UserPrefs::Get(context)) {
-            GetBraveShieldsRemote(rfh)->SetReduceLanguageEnabled(
-                brave_shields::IsReduceLanguageEnabledForProfile(pref_service));
-          }
-        }
-      });
+  RenderFrameHost* rfh = navigation_handle->GetRenderFrameHost();
```

Comment:
Looks like `navigation_handle->HasCommitted()` and `!navigation_handle->IsSameDocument()` should be checked

Reply:
added `navigation_handle->IsSameDocument()` check.

`navigation_handle->HasCommitted()` doesn't make sense in `ReadyToCommitNavigation`